src/interface/types.rs

```diff
@@ -541,6 +563,24 @@ pub fn duration_fromtimeval(union_argument: Arg) -> Result<Option<interface::Rus
     }
 }
 
+pub fn get_itimerval<'a>(union_argument: Arg) -> Result<Option<&'a mut ITimerVal>, i32> {
+    let pointer = unsafe{union_argument.dispatch_structitimerval};
+    if !pointer.is_null() {
+        Ok(Some(unsafe{&mut *pointer}))
+    } else {
+        Ok(None)
+    }
+}
+
+pub fn get_constitimerval<'a>(union_argument: Arg) -> Result<Option<&'a ITimerVal>, i32> {
+    let pointer = unsafe{union_argument.dispatch_conststructitimerval};
+    if !pointer.is_null() {
+        Ok(Some(unsafe{& *pointer}))
+    } else {
+        Ok(None)
+    }
+}
+
 pub fn duration_fromtimespec(union_argument: Arg) -> Result<interface::RustDuration, i32> {
```

Comment:
When you have an if block that ends with a return statement, you don't need to use an else block after it. The reason is that if the condition in the if block is true, the function will return, and any code after the if block won't be executed. If the condition is false, the function will continue executing the code after the if block. So, the else is redundant.

In both functions, the else block after the if isn't necessary. Since you're returning from inside the if block, you can just have the Ok(None) as the next statement.